pkg/vault/contracts/WrappedBalancerPoolTokenFactory.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import {
+    IWrappedBalancerPoolTokenFactory
+} from "@balancer-labs/v3-interfaces/contracts/vault/IWrappedBalancerPoolTokenFactory.sol";
+
+import { WrappedBalancerPoolToken } from "./WrappedBalancerPoolToken.sol";
+
+/// @notice Factory contract for creating wrapped Balancer pool tokens
+contract WrappedBalancerPoolTokenFactory is IWrappedBalancerPoolTokenFactory {
+    IVault internal immutable _vault;
+    mapping(address => address) internal _wrappedTokens;
+
+    constructor(IVault vault_) {
+        _vault = vault_;
+    }
+
+    /// @inheritdoc IWrappedBalancerPoolTokenFactory
+    function createWrappedToken(address bpt) external returns (address) {
+        address wrappedToken = _wrappedTokens[bpt];
+        if (wrappedToken != address(0)) {
+            revert WrappedBPTAlreadyExists(wrappedToken);
+        }
+
+        if (!_vault.isPoolInitialized(address(bpt))) {
```

Comment:
Could also check `isPoolRegistered` to make sure it's even a valid BPT for this Vault:

```
if (_vault.isPoolRegistered(bpt) == false) {
    revert TokenIsNotBPT();
} else if (_vault.isPoolInitialized(bpt) == false) {
    revert BalancerPoolTokenNotInitialized();
}
```
Also, the readability convention is "x == false" vs. "!x".

Reply:
Registration is a necessary condition for initialization, so this is a bit redundant if it's just to add a different error message.
I'd just check initialization.